Sources/TuistCache/ContentHashing/GraphContentHasher.swift

```diff
@@ -2,59 +2,99 @@ import Checksum
 
-    public init(fileHandler: FileHandling = FileHandler.shared) {
-        self.fileHandler = fileHandler
+    public init(
+        contentHasher: ContentHasher = ContentHasher(),
+        coreDataModelsContentHasher: CoreDataModelsContentHasher = CoreDataModelsContentHasher()
+    ) {
+        self.contentHasher = contentHasher
+        self.coreDataModelsContentHasher = coreDataModelsContentHasher
     }
 
     public func contentHashes(for graph: Graphing) throws -> [TargetNode: String] {
         let hashableTargets = graph.targets.filter { $0.target.product == .framework }
-        let hashes = try hashableTargets.map { try makeContentHash(of: $0) }
+        let hashes = try hashableTargets.map { try hash(targetNode: $0) }
         return Dictionary(uniqueKeysWithValues: zip(hashableTargets, hashes))
     }
 
-    private func makeContentHash(of targetNode: TargetNode) throws -> String {
-        // TODO: extend function to consider build settings, compiler flags, dependencies, and a lot more
-        let sourcesHash = try hashSources(of: targetNode)
-        let productHash = try hash(string: targetNode.target.productName)
-        let platformHash = try hash(string: targetNode.target.platform.rawValue)
-        return try hash(strings: [sourcesHash, productHash, platformHash])
-    }
+    // MARK: - Private
+
+    private func hash(targetNode: TargetNode) throws -> String {
+        let target = targetNode.target
+        let sourcesHash = try hash(sources: target.sources)
+        let resourcesHash = try hash(resources: target.resources)
+        let coreDataModelHash = try coreDataModelsContentHasher.hash(coreDataModels: target.coreDataModels)
+        let targetActionsHash = try hash(targetActions: target.actions)
+        let stringsToHash = [sourcesHash,
+                             target.name,
+                             target.platform.rawValue,
+                             target.product.rawValue,
+                             target.bundleId,
+                             target.productName,
+                             resourcesHash,
+                             coreDataModelHash,
+                             targetActionsHash]
+
+
+        //TODO: hash headers, platforms, version, entitlements, info.plist, target.environment, target.filesGroup, targetNode.settings, targetNode.project, targetNode.dependencies ,targetNode.target.dependencies
 
-    private func hashSources(of targetNode: TargetNode) throws -> String {
-        let hashes = try targetNode.target.sources.sorted(by: { $0.path < $1.path }).map(md5)
-        let joinedHash = try hash(strings: hashes)
-        return joinedHash
+        return try contentHasher.hash(stringsToHash)
     }
 
-    private func hash(string: String) throws -> String {
-        guard let hash = string.checksum(algorithm: .md5) else {
-            throw ContentHashingError.stringHashingFailed(string)
+    private func hash(headers: Headers) throws -> String {
```

Comment:
It's a WIP, but the idea is to extract all these private functions into objects, like I did for `CoreDataModelsContentHasher`. What do you think @pepibumur ? I want to get a feedback before doing too much work

Reply:
I think it's a good idea. We have done that with the generation logic where we have classes like `BuildPhaseGenerator`, `TargetGenerator`... That encourages to have small and self-contained classes, instead of massive ones, full of private methods. It makes testing them in isolation easier too.